src/libertem/executor/pipelined.py

```diff
@@ -128,22 +166,31 @@ def kill_worker(self, worker_info: PoolWorkerInfo, timeout: float = 5.0):
             worker_info.process.join(30)
 
     def kill(self, timeout: float = 5):
+        if self._workers is None:
+            raise PoolStateError("Cannot kill workers, not running")
         for worker in self._workers:
             self.kill_worker(worker, timeout=timeout)
         exitcodes = [
             worker.process.exitcode
             for worker in self._workers
         ]
         self._workers = None
+        if self._response_q is None:
+            raise PoolStateError("Response queue not available")
         self._response_q.close(drain=False, force=True)
         self._response_q = None
-        assert all([e is not None for e in exitcodes])
+        if any([e is None for e in exitcodes]):
+            raise PoolStateError("At least one worker failed to be killed")
 
     @property
     def workers(self) -> List[PoolWorkerInfo]:
+        if self._workers is None:
```

Comment:
Is it important that this raises? I would expect `self._workers is None` to return `[]`. Seems to match the 'somewhat permissive' form of `all_alive(self)`.